src/when.test.js

```diff
@@ -232,6 +232,78 @@ describe('When', () => {
       expect(() => fn(doNotCallMeBro)).not.toThrow()
     })
 
+    describe('when.allArgs', () => {
+      it('throws an error if you try to use other matches with it', () => {
+        const fn = jest.fn()
+
+        when(fn)
+          .calledWith(when.allArgs(() => true), 1, 2, 3)
+          .mockReturnValue('x')
+
+        expect(() => fn(3, 6, 9)).toThrow(/When using when.allArgs, it must be the one and only matcher provided to calledWith. You have incorrectly provided other matchers along with when.allArgs./)
+      })
+
+      it('allows matching against all the args at once with when.allArgs', () => {
+        const fn = jest.fn()
+
+        const numberDivisibleBy3 = (args) => args.every(arg => arg % 3 === 0)
+
+        when(fn)
+          .calledWith(when.allArgs(numberDivisibleBy3))
+          .mockReturnValue('x')
+
+        expect(fn(3, 6, 9)).toEqual('x')
+        expect(fn(3, 6, 10)).toBeUndefined()
+        expect(fn(1, 2, 3)).toBeUndefined()
+      })
+
+      it('react use case from github', () => {
+        // SEE: https://github.com/timkindberg/jest-when/issues/66
+
+        const SomeChild = jest.fn()
+
+        when(SomeChild)
+          .calledWith({ xyz: '123' })
+          .mockReturnValue('hello world')
+
+        const propsOf = propsToMatch => when.allArgs(([props, refOrContext], equals) => equals(props, propsToMatch))
+
+        when(SomeChild)
+          .calledWith(propsOf({ xyz: '123' }))
+          .mockReturnValue('hello world')
+
+        expect(SomeChild({ xyz: '123' })).toEqual('hello world')
+      })
```

Comment:
@mcous ^^

Reply:
I think this `propsOf` utility is how we'd end up actually using this, so this test case looks solid to me